From 8dbb135c14392e61111a4dfcceffe990148b68fd Mon Sep 17 00:00:00 2001
From: Juergen Repp <juergen_repp@web.de>
Date: Sat, 8 Feb 2025 19:59:01 +0100
Subject: [PATCH 7/8] tpm2_eventlog_yaml.c Fix output of BlobDescription.

The last byte of the BlobDescription for an EV_EFI_PLATFORM_FIRMWARE_BLOB2
event was not displayed.
Fixes: #3455.

Signed-off-by: Juergen Repp <juergen_repp@web.de>
---
 lib/tpm2_eventlog_yaml.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/tpm2_eventlog_yaml.c b/lib/tpm2_eventlog_yaml.c
index 59a5d8fc..343fb658 100644
--- a/lib/tpm2_eventlog_yaml.c
+++ b/lib/tpm2_eventlog_yaml.c
@@ -582,7 +582,7 @@ bool yaml_uefi_platfwblob2(UEFI_PLATFORM_FIRMWARE_BLOB2 *data) {
     return false;
   }
 
-  bytes_to_str (data->BlobDescription, blobdescsize, eventdesc, 2*blobdescsize);
+  bytes_to_str (data->BlobDescription, blobdescsize, eventdesc, 2*blobdescsize + 1);
 
   tpm2_tool_output("  Event:\n"
                    "    BlobDescriptionSize: %d\n"
-- 
2.49.0

